target_stitch/__init__.py

```diff
@@ -541,26 +541,31 @@ def __init__(self, # pylint: disable=too-many-arguments
 
 
 
-    def flush(self):
+    def flush_stream(self, messages):
         '''Send all the buffered messages to Stitch.'''
 
-        if self.messages:
-            stream_meta = self.stream_meta[self.messages[0].stream]
-            for handler in self.handlers:
-                handler.handle_batch(self.messages,
-                                     self.contains_activate_version,
-                                     stream_meta.schema,
-                                     stream_meta.key_properties,
-                                     stream_meta.bookmark_properties,
-                                     self.state_writer,
-                                     self.state)
-
-            self.time_last_batch_sent = time.time()
-            self.messages = []
-            self.contains_activate_version = False
-            self.state = None
-            self.buffer_size_bytes = 0
+        stream = messages[0].stream
+        stream_meta = self.stream_meta[stream]
+        for handler in self.handlers:
+            handler.handle_batch(messages,
+                                 self.contains_activate_version.get(stream, False),
+                                 stream_meta.schema,
+                                 stream_meta.key_properties,
+                                 stream_meta.bookmark_properties,
+                                 self.state_writer,
+                                 self.state)
+
+        self.time_last_batch_sent = time.time()
+        self.contains_activate_version[stream] = False
+        self.state = None
+        self.buffer_size_bytes = 0
+
 
+    def flush(self):
+        if sum([len(ms) for ms in self.messages.values()]) > 0:
+            for messages in self.messages.values():
+                self.flush_stream(messages)
+            self.messages = {k: [] for k in self.messages.keys()}
```

Comment:
Maybe this should be moved per-stream into flush_stream for consistency, and to keep all the resets together.

Reply:
Done